subsys/fs/littlefs_fs.c

```diff
@@ -600,8 +619,49 @@ static int littlefs_mount(struct fs_mount_t *mountp)
 	dev = flash_area_get_device(fs->area);
 	if (dev == NULL) {
 		LOG_ERR("can't get flash device: %s", log_strdup(fs->area->fa_dev_name));
-		ret = -ENODEV;
-		goto out;
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int littlefs_mount(struct fs_mount_t *mountp)
+{
+	int ret = 0;
+	struct fs_littlefs *fs = mountp->fs_data;
+	bool block_dev = littlefs_on_blkdev(mountp);
+
+	LOG_INF("LittleFS version %u.%u, disk version %u.%u",
+		LFS_VERSION_MAJOR, LFS_VERSION_MINOR,
+		LFS_DISK_VERSION_MAJOR, LFS_DISK_VERSION_MINOR);
+
+
+	if (block_dev) {
```

Comment:
```if (IS_ENABLED(...) && block_dev)```

Reply:
Ok, so you want to have cached 'block_dev' bool variable and the `IS_ENABLED()` macro expansion in each single `if`.
I will change that (but personally I think that `IS_ENABLED()` here is an overkill).

Reply:
But it is in your code.
If you do the
```
if (IS_ENABLED(...) && block_dev) {
...
}
````
This is should be exactly, except for compilation phase, as your code
```
if (block_dev) {
#if IS_ENABLED(CONFIG_FS_LITTLEFS_BLK_DEV)
...
#endif
}
```

Reply:
@de-nordic 
No, this would not be the same.

The subtle difference is that I would need to provide the code which is inside:
```
if (IS_ENABLED(...) && block_dev) {
...
}
```
to make the file compile. For example you would need to always add `disk_*() API functions` to the build even when you don't use it.

With 
```
if (block_dev) {
#if IS_ENABLED(CONFIG_FS_LITTLEFS_BLK_DEV)
...
#endif
}
```
The stuff between `#if #endif` will be not compiled at all, as being removed by the compiler.

Reply:
s/compiler/preprocessor

Reply:
move `IS_ENABLED` to `if (block_dev)`:
```
if (IS_ENABLED(CONFIG_FS_LITTLEFS_BLK_DEV) && block_dev)
```

do it everywhere where `IS_ENABLED` can be merged with `if`